holoviews/element/raster.py

```diff
@@ -408,45 +211,157 @@ class Image(SheetCoordinateSystem, Raster):
     access to the data, via the .data attribute.
     """
 
-    bounds = param.ClassSelector(class_=BoundingRegion, default=BoundingBox(), doc="""
+    bounds = param.ClassSelector(class_=BoundingRegion, default=None, doc="""
        The bounding region in sheet coordinates containing the data.""")
 
+    datatype = param.List(default=['image', 'grid', 'xarray', 'dataframe', 'dictionary'])
+
     group = param.String(default='Image', constant=True)
 
+    kdims = param.List(default=[Dimension('x'), Dimension('y')],
+                       bounds=(2, 2), constant=True, doc="""
+        The label of the x- and y-dimension of the Raster in form
+        of a string or dimension object.""")
+
     vdims = param.List(default=[Dimension('z')],
-                       bounds=(1, 1), doc="""
+                       bounds=(1, None), doc="""
         The dimension description of the data held in the matrix.""")
 
-
     def __init__(self, data, bounds=None, extents=None, xdensity=None, ydensity=None, **params):
-        bounds = bounds if bounds is not None else BoundingBox()
-        if np.isscalar(bounds):
+        extents = extents if extents else (None, None, None, None)
+        if data is None: data = np.array([[0]])
+        Dataset.__init__(self, data, extents=extents, bounds=None, **params)
+
+        if isinstance(data, np.ndarray):
+            dim2, dim1 = data.shape[:2]
+        else:
+            dim1, dim2 = tuple([len(self.dimension_values(kd, expanded=False))
+                                  for kd in self.kdims])
+        if self.bounds is not None:
+            bounds = self.bounds
+        elif bounds is None and isinstance(data, np.ndarray):
+            bounds = BoundingBox()
+
+        if bounds is None:
+            xvals = self.dimension_values(0, False)
+            l, r, xdensity, _ = util.bound_range(xvals, xdensity)
+            yvals = self.dimension_values(1, False)
+            b, t, ydensity, _ = util.bound_range(yvals, ydensity)
+            bounds = BoundingBox(points=((l, b), (r, t)))
```

Comment:
Not too bad. I assume we can't just get it from the interface as that would mean accessing special API unique to Image?